BrainPortal/app/views/tool_configs/_form_fields.html.erb

```diff
@@ -212,17 +212,20 @@
       </div>
     <% end %>
 
-    <% t.edit_cell(:singularity_overlays_specs, :header => "Singularity Overlays", :show_width => 2) do |f| %>
-      <%= f.text_area :singularity_overlays_specs, :rows => 4, :cols => 120 %>
+    <% t.edit_cell(:singularity_overlays_specs, :header => "Singularity Overlays", :show_width => 2, :content => full_description(@tool_config.singularity_overlays_specs)) do |f| %>
+      <%= f.text_area :singularity_overlays_specs, :rows => 6, :cols => 120 %>
         <div class="wide_field_explanation">
         This field can contain one or several specifications for data overlays
         to be included when the task is started with Singularity.
         A specification can be either
-        a full path (e.g. <em>/a/b/data.squashfs</em>),
-        a path with a pattern (e.g. <em>/a/b/data*.squashfs</em>),
+        a full path (e.g. <em>file:/a/b/data.squashfs</em>),
+        a path with a pattern (e.g. <em>file:/a/b/data*.squashfs</em>),
+        a registered file identified by ID (e.g. <em>userfile:123</em>)
         or a SquashFS Data Provider identified by its ID or name (e.g. <em>dp:123</em>, <em>dp:DpNameHere</em>).
         In the case of a Data Provider, the overlays will be the files that the provider uses.
-        Separate specifications with blank, commas, or newlines.
+        Each overlay specification should be on a separate line.
+        You can add comments, indicated with hash symbol <em>#</em>.
+        For example, <em>file:/a/b/atlas.squashfs # brain atlas</em>
```

Comment:
Your parser doesn't seem to support comments on the same line as a spec... ? Either adjust the parser or remove the suggestion here.

Reply:
That's only comment I am not agree. Same line comments are supported. Worked on my tests. But provide your examples, if you feel some corner case is missed.

Reply:
Okay, I didn't see the full effect of the split() in parsed_overlay_specs()